app/components/layout/sidebar/sidebar.tsx

```diff
@@ -11,12 +11,56 @@ import SidebarBottom from "./bottom";
 import MenuButton from "./menu-button";
 import MenuItems from "./menu-items";
 import Overlay from "./overlay";
+import { useMediaStream } from "~/components/scan-qr/media_stream_provider";
+import QRCodeScanner from "~/components/scan-qr/QRCodeScanner";
 
 export default function Sidebar() {
   const { user, minimizedSidebar } = useLoaderData<typeof loader>();
   const [isMobileNavOpen, toggleMobileNav] = useAtom(toggleMobileNavAtom);
   const mainNavigationRef = useRef<HTMLElement>(null);
 
+  const [showScanner, setShowScanner] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [scannedSuccessfully, setScannedSuccessfully] = useState(false);
+
+  const { startMediaStream, stopMediaStream } = useMediaStream();
+
+  const isValidQRCode = (data: string) => {
+    const regex = /^(https?:\/\/)([^/:]+)(:\d+)?\/qr\/[a-zA-Z0-9]+$/;
+    return regex.test(data);
+  };
+
+  const handleQRScan = (data: string | null) => {
+    if (data) {
+      if (isValidQRCode(data)) {
+        setScannedSuccessfully(true);
```

Comment:
On a successful scan, you need to navigate the user to the route.
To do that you can do the following:
```ts
const navigate = useNavigate()
navigate(`/qr/${qr.id}`)
```